### ISSUE TITLE
Instructions for updating are wrong

### ISSUE THREAD

**Issue by brianreinhold** [angular/angular#59037]
### Describe the problem that you experienced
**ERROR 1**
This page
https://angular.dev/update-guide?v=18.0-19.0&l=2
Supposedly tells you how to migrate from Angular 18 to 19.
I follow the instructions and immediately get an error
```
Updating multiple major versions of '@angular/core' at once is not supported. Please migrate each major version individually.
Run 'ng update @angular/core@18' in your workspace directory to update to latest '18.x' version of '@angular/core'.
```
Well, that was never mentioned, So I do as requested and get this error

```Fetching dependency metadata from registry...
                  Package "@angular-devkit/build-angular" has an incompatible peer dependency to "typescript" (requires ">=5.2 <5.3", would install "5.5.4").
✖ Migration failed: Incompatible peer dependencies found.
Peer dependency warnings when installing dependencies means that those dependencies might not work correctly together.
You can use the '--force' option to ignore incompatible peer dependencies and instead address these warnings later.
  See "C:\Users\BRIANR~1\AppData\Local\Temp\ng-XKJ8EJ\angular-errors.log" for further details.
```
What is missing in these update instructions? Do I need to delete the `package-lock.json` file?

At this point, have I messed up my original project?

**ERROR 2**
Here is another issue. I ran
`ng update @angular/cli@18 @angular/core@18`

and it reported the following
```
    Updating package.json with dependency @angular-devkit/build-angular @ "18.2.12" (was "17.0.0")...
    Updating package.json with dependency @angular/cli @ "18.2.12" (was "17.3.7")...
    Updating package.json with dependency @angular/compiler-cli @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency typescript @ "5.5.4" (was "5.2.2")...
    Updating package.json with dependency @angular/animations @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/common @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/compiler @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/core @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/forms @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/platform-browser @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/platform-browser-dynamic @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency @angular/router @ "18.2.13" (was "17.0.2")...
    Updating package.json with dependency zone.js @ "0.14.10" (was "0.14.2")...
```
All the 'was' statements are wrong. Doing a simple diff in git shows it was 18.2.5 previously. What is going on? Where did that 17 come from?

**ERROR 3**
Another error. After reverting the changes in git because one was forced to run a skippable item (so I did a ctrl-C) running

`ng update @angular/cli@18 @angular/core@18`
gave
`Error: Repository is not clean. Please commit or stash any changes before updating.`

Which is an error. The changes were reverted and there is nothing to commit. Now the API is stuck. Not sure how to do anything more.

If the instructions said
run
ng update @angular/cli@18 @angular/core@18
first (I deleted the lock file and committed - not sure if that is necessary) the rest of the instructions appear to work

**----------------------- From here one refers to ERROR 1**

### Enter the URL of the topic with the problem

https://angular.dev/update-guide?v=18.0-19.0&l=2

### Describe what you were looking for in the documentation

Instructions which, when followed, would not generate errors (because the instructions are wrong or have left out important steps).

### Describe the actions that led you to experience the problem

Trying to migrate from Angular 18 to Angular 19

### Describe what you want to experience that would fix the problem

That following the update instructions would work and I would migrate my 18 project to 19

### If this problem caused an exception or error, please paste it here

```true

```

### If the problem is browser-specific, please specify the device, OS, browser, and version

```true

```

### Provide any additional information here in as much as detail as you can

```true

```

**Comment by wartab** [comment]
Angular checks what version you actually have installed when running the update scripts. Your node_modules were probably not up to date with what's actually in your dependency list. You were probably still running Angular with version 17 without realising.

**Comment by JeanMeche** [comment]
Hello, we reviewed this issue and determined that it doesn't fall into the bug report or feature request category. This issue tracker is not suitable for support requests, please repost your issue on [StackOverflow](https://stackoverflow.com/) using tag `angular`.

If you are wondering why we don't resolve support issues via the issue tracker, please [check out this explanation](https://github.com/angular/angular/blob/main/CONTRIBUTING.md#question).

**Comment by brianreinhold** [comment]
@JeanMeche If the instructions indicated that one should do 
ng update @angular/cli@18 @angular/core@18
first, the series of errors would not have occurred (I believe).

**Comment by JeanMeche** [comment]
You're issue is that you did not update v18 app but a v17.

**Comment by brianreinhold** [comment]
@wartab I would hope not since I deleted the directory and did an npm install with the package.json files updated. But I have to admit, you have to dig deep into the modules repo to find out what versions are actually installed (which I did not do as I assumed the update commands worked as advertised).

**Comment by brianreinhold** [comment]
@JeanMeche I had done a similar update from 17 to 18 a few months ago. I followed the instructions and recall no problem. All my json files got updated to 18 so I never questioned the uploaded modules. Maybe they didn't get updated to 18. I'll never know since I do not commit those files but simply use npm install and let the package.json files do their thing when I install the project on another computer.
